cli/src/commands/prev.rs

```diff
@@ -79,24 +77,15 @@ pub(crate) fn cmd_prev(
             .heads()
             .contains(current_wc_id);
     let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    let start_id = if edit {
-        current_wc_id
-    } else {
-        match current_wc.parent_ids() {
-            [parent_id] => parent_id,
-            _ => return Err(user_error("Cannot run `jj prev` on a merge commit")),
-        }
-    };
-    let ancestor_expression = RevsetExpression::commit(start_id.clone()).ancestors_at(offset);
+    // If --edit is specified or implied, then the offset is relative to the
+    // current working copy commit. Otherwise, the offset is relative to the
+    // parent(s) of the current wc commit.
     let target_revset = if edit {
-        ancestor_expression
+        RevsetExpression::commit(current_wc_id.clone()).ancestors_at(args.offset)
```

Comment:
The documentation for `ancestors_at` seems wrong:

```
    /// Ancestors of `self`, including `self` until `generation` back.
    pub fn ancestors_at(self: &Rc<RevsetExpression>, generation: u64) -> Rc<RevsetExpression> {
        self.ancestors_range(generation..(generation + 1))
    }
 ```
Based on the docstring the `target_revset` should include the wc commit and all of the ancestors back to the specified offset. Instead it looks like `ancestors_at` actually returns only the ancestors at the specified generation number.

I think this is either a bug in the documentation or the implementation. Perhaps the docs are wrong?

@yuja

Reply:
Yes, the doc is wrong. As the function name suggests, it should return `generation`-th ancestors counted from the `self`.

Reply:
Sounds good. Suggesting an edit in https://github.com/martinvonz/jj/pull/3471